src/EPR.PRN.Backend.API/Services/PrnService.cs

```diff
@@ -93,7 +92,32 @@ public async Task UpdateStatus(Guid orgId, Guid userId, List<PrnUpdateStatusDto>
         }
         await _repository.SaveTransaction(transaction);
     }
+    public async Task InsertPeprNpwdSyncPrns(List<InsertSyncedPrn> syncedPrns)
+    {
+        List<Eprn> prns = await _repository.GetPrnsForPrnNumbers(syncedPrns.Select(p => p.EvidenceNo).ToList());
+        var nonExistingPrns = syncedPrns.Select(x => x.EvidenceNo).Except(prns.Select(x => x.PrnNumber));
```

Comment:
I would expect a foreign key to handle this automatically

Reply:
If we rely on foreign key we need to intercept db error and then get the non existing one. Since we have all info here did with linq